libs/ui-components/src/hooks/useDeviceLabelMatch.ts

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+
+import { Fleet, FleetList } from '@flightctl/types';
+import { FlightCtlLabel } from '../types/extraTypes';
+import { useFetch } from './useFetch';
+import { getErrorMessage } from '../utils/error';
+
+export type DeviceMatchStatus = {
+  status:
+    | 'unchecked'
+    | 'unchecked--invalid'
+    | 'checking'
+    | 'checked--error'
+    | 'checked--unique'
+    | 'checked--multiple'
+    | 'checked--empty';
+  detail?: string;
+};
+
+type MatchLabelsFn = (labels: FlightCtlLabel[], hasErrors: boolean) => void;
+
+const getMatchResult = (fleets: Fleet[], deviceLabels: FlightCtlLabel[]): DeviceMatchStatus => {
+  const matchingFleets = fleets.filter((fleet) => {
+    const fleetMatch = fleet.spec.selector?.matchLabels || {};
+    return Object.entries(fleetMatch).every(([fleetMatchKey, fleetMatchValue]) => {
+      const matchingDeviceLabel = deviceLabels.find((dLabel) => dLabel.key === fleetMatchKey);
+      return matchingDeviceLabel && (matchingDeviceLabel.value || '') === (fleetMatchValue || '');
+    });
+  });
+
+  const matchCount = matchingFleets?.length || 0;
+
+  switch (matchCount) {
+    case 0:
+      return { status: 'checked--empty' };
+    case 1:
+      return { status: 'checked--unique', detail: matchingFleets[0].metadata.name };
+    default:
+      return { status: 'checked--multiple' };
+  }
+};
+
+const useDeviceLabelMatch = (): [MatchLabelsFn, DeviceMatchStatus] => {
+  const { get } = useFetch();
+  const currentErrorRef = React.useRef<boolean>();
```

Comment:
Can you explain why we need this ? We never look at the actual value in the code, so I am confused

Reply:
Bad copy paste, I shouldn't have included `currentErrorRef`. I'll remove it.